src/engine/events.rs

```diff
@@ -1345,4 +1346,91 @@ where
         .await;
         result
     }
+
+    /// Provides an abstracted database rel read operation using warpgrapher inputs. This is the
+    /// recommended way to read relationships in a database-agnostic way that ensures the event handlers
+    /// are portable across different databases.
+    ///
+    /// # Arguments
+    ///
+    /// * `src_node_label` - String reference represing name of node type (ex: "User").
+    /// * `rel_label` - String reference representing the name of the relationship (ex: "teams").
+    /// * `input` - `Value` describing the relationship query input.
+    /// * `partition_key_opt` - Optional `Value` describing the partition key if the underlying database supports it.
+    ///
+    /// # Examples
+    ///
+    /// ```rust, no_run
+    /// # use serde_json::json;
+    /// # use warpgrapher::Error;
+    /// # use warpgrapher::engine::events::EventFacade;
+    /// # use warpgrapher::engine::objects::Rel;
+    /// # use warpgrapher::engine::value::Value;
+    /// # use warpgrapher::juniper::BoxFuture;
+    ///
+    /// fn before_user_read(value: Value, mut ef: EventFacade<()>) -> BoxFuture<Result<Value, Error>> {
+    ///     Box::pin(async move {
+    ///         let rels: Vec<Rel()> = ef.readl_rels(
```

Comment:
Typo here, I think. `readl_rels` should probably be `read_rels`.

Reply:
Good catch.